### ISSUE TITLE
Magellan in Internet Explorer anchor links don't work

### ISSUE THREAD

**Issue by jvanv** [zurb/foundation-sites#8147]
Apologies if this has been reported, but noticed when browsing http://foundation.zurb.com/sites/docs/magellan.html on a Windows 7 virtual machine using IE10 that the links in all the Magellan navigation elements don't work, no scroll and no default jump behaviour, links just do nothing.

Haven't tried it on other versions of IE

Foundation version 6 (whatever is on http://foundation.zurb.com/sites/docs/magellan.html)

**Comment by manolenso** [comment]
Used magellan lib 6.1.2 ?

**Comment by jvanv** [comment]
Using that on the project which made me notice this, from bower.json:

  "version": "6.1.2",

When testing that project the anchors don't work in ie9 through to ie11. Sorry I can't share the development url. I was debugging it when I noticed your docs site has the same issue so thought I'd report it. We've done a sort of hack around it, which I will post here for posterity :)

`//scroll for ie
function scrollToAnchor(aid){
    var aTag = jQuery(aid);
    jQuery('html,body').animate({scrollTop: aTag.offset().top},'slow');
}

function isIE() {

	var ua = window.navigator.userAgent;
	var msie = ua.indexOf("MSIE ");

	if (msie > 0 || !!navigator.userAgent.match(/Trident.*rv\:11\./))      // If Internet Explorer, return version number
		return true;
	else                 // If another browser, return 0
		return false;

	return false;
}

if(isIE()) {
	jQuery('.menu li a').click(function(){
		scrollToAnchor(jQuery(this).attr('href'));
	});
}`

**Comment by manolenso** [comment]
Thank for the hack, use the 6.1.1  instead the last!

**Comment by manolenso** [comment]
look at the #8138

**Comment by jvanv** [comment]
Nice one, I thought I couldn't have been the only one to notice this. We'll leave the hack in place, it works OK for now and we'll update assuming this is fixed in the next version :)

**Comment by manolenso** [comment]
Cool simplified your hack
`function scrollToAnchor(aid){
    var aTag = jQuery(aid);
    jQuery('html,body').animate({scrollTop: aTag.offset().top},'slow');
}
    jQuery('.menu li a').click(function(){
        scrollToAnchor(jQuery(this).attr('href'));
    });
That's ok for IE, Crhome, Firefox`

**Comment by jvanv** [comment]
Thanks, just updated it for Firefox, will give yours a go too, thanks :)

```
//scroll for ie & firefox
function scrollToAnchor(aid){
    var aTag = jQuery(aid);
    jQuery('html,body').animate({scrollTop: aTag.offset().top},'slow');
}

function isIE() {

	var ua = window.navigator.userAgent;
	var msie = ua.indexOf("MSIE ");

	if (msie > 0 || !!navigator.userAgent.match(/Trident.*rv\:11\./))      // If Internet Explorer, return version number
		return true;
	else                 // If another browser, return 0
		return false;

	return false;
}

var is_firefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;

if(isIE() || (is_firefox)) {
	jQuery('.menu li a').click(function(){
		scrollToAnchor(jQuery(this).attr('href'));
	});
}
```

**Comment by colin-marshall** [comment]
@jvanv @manolenso is this still an issue with the latest versions of Foundation?

**Comment by JeremyEnglert** [comment]
Looks like this was resolved in 6.2.0.

However, if the issue still exists, feel free to reopen this ticket using the new "Issues Template: https://github.com/zurb/foundation-sites/blob/develop/.github/ISSUE_TEMPLATE.md

Big thanks!
